### DISCUSSION TITLE
Best approach - thumbnail generator

### DISCUSSION THREAD

**Discussion by AlexandreLrz** [bghira/SimpleTuner#1761] [Q&A]
Hi everyone,

I'm trying to create a YouTube thumbnail generator using Hidream, and for that, I'm experimenting a lot with different datasets and hyperparameters. I still cannot get something satisfying even after tens of runs, which is why I'm wondering if you have some feedback from your previous work that may help me move in the right direction for this project.

The main difficulty I have is learning multiple styles (entertainment, business, gaming, etc.). I have about 8 styles to learn, and I'm not sure yet what to do: either teach one single model to learn all the different styles using a bigger rank like 64, or teach multiple smaller LoRAs/LoKRs with a lower rank like 16.

Here is what I tried so far:

## Algo
- **Lora**: I had the best results from Lora using the lycoris config (rank 16)
- **Lokr**: When using LoKR, for some reason, I struggled to learn anything. When I increased the learning rate with LoKR, I saw the beginning of a style emerging, but lots of artifacts appeared at the same time (6 fingers, deformed heads, etc.). I'm still not sure if I should experiment more with LoKR, as I've heard it should be the best choice. One thing I still don't understand with LoKR is the number of parameters decreasing when increasing the factor. I understand it's inversely proportional to the number of parameters, but in this case, isn't it better to decrease the factor to learn more difficult concepts (isn't more parameters = better at learning?).

## Dataset
I tried three differents parts of my dataset :
- **Entire dataset** (20k images): this gave me the worst results. From time to time, I successfully learned one or two styles from niches, but after a few more steps, it got worse.
- **Single style** (700 images): I successfully learned the style of one single niche using only the pictures from this style, but the placement of objects/people was still wrong, and I was getting artifacts after 2k steps.
- Single style (1 image): Unexpectedly, I had the best results when training on a single image. No artifacts, and I was getting closer and closer to the desired style for most of my validation samples (placement was still a problem).

## Steps
Since I'm running my tests on Runpod and it's quite expensive, I stop any run when it seems to get worse or stops learning (maybe I should continue?).
- For [full dataset] I ran about 10k steps (batch size=2) and saw almost no difference from the default model (1-2 styles appeared then disappeared)
- For [700 pictures] I ran 2k steps (batch size=4) and started to see my style coming in at 1500
- For [1 picture] I ran 5k steps and started to see my style coming at 2000

## Loss
- The loss was decreasing only when I was using a single picture in the training set.
<img width="365" height="311" alt="Capture d’écran 2025-10-05 à 22 05 28" src="https://github.com/user-attachments/assets/ddd7fadd-6c68-4889-bea5-85ac2079593e" />

- When I used more (700 or 20k), the EMA loss didn’t really decrease (or maybe too slowly to be visible).
<img width="368" height="317" alt="Capture d’écran 2025-10-05 à 22 05 20" src="https://github.com/user-attachments/assets/15efbb13-5a72-4210-9d69-121dff2eaf1e" />

---

I'm probably still doing something wrong, as I'm really far from what I'm expecting, maybe too few steps...

Thank you in advance for any help regarding this.
